src/mod/endpoints/mod_sofia/mod_sofia.c

```diff
@@ -539,10 +539,10 @@ switch_status_t sofia_on_hangup(switch_core_session_t *session)
 				if ((switch_channel_test_flag(channel, CF_INTERCEPT) || cause == SWITCH_CAUSE_PICKED_OFF || cause == SWITCH_CAUSE_LOSE_RACE)
 					&& !switch_true(switch_channel_get_variable(channel, "ignore_completed_elsewhere"))) {
 					reason = switch_core_session_strdup(session, "SIP;cause=200;text=\"Call completed elsewhere\"");
-				} else if (cause > 0 && cause < 128) {
+				} else if ((cause > 0 && cause < 128) || switch_telnyx_is_custom_q850_code(cause)) {
```

Comment:
Those custom Q850 errors are both from FS and Telnyx. Ideally this function to check for customq850 can be a part of switch_channel instead.